NachoSpigot-Server/src/main/java/net/minecraft/server/EntityHuman.java

```diff
@@ -1014,34 +1014,25 @@ public void attack(Entity entity) {
 
                     if (flag2) {
                         if (i > 0) {
-                            KnockbackConfig config = Knockback.get().getConfig();
-                            if(!config.customKnockback) entity.g(-MathHelper.sin(this.yaw * 3.1415927F / 180.0F) * (float) i * 0.5F, 0.1D, (double) (MathHelper.cos(this.yaw * 3.1415927F / 180.0F) * (float) i * 0.5F)); else
+                            KnockbackProfile profile = (entity.getKnockbackProfile() == null) ?
+                                    Nacho.get().getKnockbackConfig().getCurrentKb() : entity.getKnockbackProfile();
                             entity.g(
-                                    (-MathHelper.sin(this.yaw * 3.1415927F / 180.0F) * (float) i * config.knockbackExtraHorizontal),
-                                    config.knockbackExtraVertical,
-                                    (MathHelper.cos(this.yaw * 3.1415927F / 180.0F) * (float) i * config.knockbackExtraHorizontal));
+                                    (-FastMath.sin(this.yaw * Math.PI / 180.0D) * i * profile.getExtraHorizontal()),
+                                   profile.getExtraVertical(),
+                                    (FastMath.cos(this.yaw * Math.PI / 180.0D) * i * profile.getExtraHorizontal()));
```

Comment:
```suggestion
                                    (-MathHelper.sin(this.yaw * Math.PI / 180.0D) * i * profile.getExtraHorizontal()),
                                   profile.getExtraVertical(),
                                    (MathHelper.cos(this.yaw * Math.PI / 180.0D) * i * profile.getExtraHorizontal()));
```
See https://github.com/CobbleSword/NachoSpigot/pull/231#issuecomment-968184311